controllers/storagecluster/storageclasses.go

```diff
@@ -228,6 +228,13 @@ func newCephBlockPoolStorageClassConfiguration(initData *ocsv1.StorageCluster) S
 		disable:           managementSpec.DisableStorageClass,
 		isClusterExternal: initData.Spec.ExternalStorage.Enable,
 	}
+
+	// In case of erasure coded, parameter `dataPool` is not empty. The parameter`dataPool` is the name of pool and `pool` is the name erasure coded pool
+	if data["dataPool"] != "" {
+		sc.storageClass.Parameters["pool"] = generateNameForErasureCodedCephBlockPool(initData)
+		sc.storageClass.Parameters["dataPool"] = generateNameForCephBlockPool(initData)
+	}
+	return sc
```

Comment:
I wonder if this block is actually needed since the cluster is external and we don't support EC in converged AFAIK.
@aruniiird to confirm?

Reply:
You are right, we don't require `dataPool` value to be set here.
IMO, we don't even have to touch the function `newCephBlockPoolStorageClassConfiguration()` at all because, as you mentioned in the above comment, that `for` loop will provide appropriate values for both `pool` and `dataPool` parameters.

Reply:
So, basically we don't need any change just adding unit test to verify working.